src/cache/cache.jl

```diff
@@ -1,3 +1,86 @@
+Base.@kwdef struct AtmosCache{
```

Comment:
Is there a reason we need `Base.@kwdef` here? i.e. do we have default values for most fields? 
Otherwise a normal `struct` should suffice!

Reply:
I used @kwdef because the order of the parameters is arbitrary and I didn't want users to have to strictly follow the order. In the `coupler_compatibility` test I use use keyword constructor.

Reply:
I do not see a specific advantage with this in terms of user convenience! 
We still need to specify (and remember to specify) all fields anyways! 
And a keyword constructor is convenient if we can specify a few and use defaults for others!

Reply:
It is not required for compatibility. I just saw it as a way to "forget about the order" of arguments. Without this constructor, we would have to provide the arguments in the correct order, and since there is no given order, it felt natural to make them keyword.

I don't have strong feelings about this and I can change it to positional only if you prefer.